src/logic/hooks/events/use-on-loaded.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Listens for the load event of the provided target element - `window` by default - & returns its loaded status.
+ * @param targetElement The target element of which to listen for its loaded status.
+ * @return The loaded status of the provided target element.
+ */
+export const useOnLoaded = (targetElement = window) => {
+  const [loaded, setLoaded] = useState(false);
+  const loadedHandler = () => setLoaded(true);
+
+  useEffect(
+    () => {
+      if (!targetElement.addEventListener) {
```

Comment:
Removing this conditional as it isn't providing a whole lot of value. I really want to check if the element supports the "onload" event, but I'm unsure how to do that, or if it's even an easy thing to do, or at all possible w/o a try/catch.